ocaml/xapi/repository_helpers.ml

```diff
@@ -1413,18 +1413,9 @@ let get_ops_of_pending ~__context ~host ~kind =
   in
   let get_pending_guidances_of_vms ~db_get =
     Db.Host.get_resident_VMs ~__context ~self:host
-    |> List.map (fun self -> (self, Db.VM.get_record ~__context ~self))
-    |> List.filter_map (fun (ref, record) ->
-           match
-             ( record.API.vM_is_control_domain
-             , record.API.vM_power_state
-             , Helpers.has_qemu_currently ~__context ~self:ref
-             )
-           with
-           | false, `Running, true | false, `Paused, true ->
-               Some ref
-           | _ ->
-               None
+    |> List.filter (fun self ->
+           (not (Db.VM.get_is_control_domain ~__context ~self))
+           && Helpers.has_qemu_currently ~__context ~self
```

Comment:
Why is the QEMU check here?

Reply:
So now all VMs should be with device models?

Reply:
Not necessarily. Is this for the RestartDevice model guidance? What about RestartVMs, which should apply to any VMs?

Reply:
Yes, this is for RestartDeviceModel.
 RestartVM does not rely on this. But it's just harmless to do it here.